components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/recovery/v1/impl/core/PasswordRecoveryService.java

```diff
@@ -407,19 +427,24 @@ private RecoveryChannelInformation buildRecoveryChannelInformation(
         return recoveryChannelInformation;
     }
 
-    /**
-     * Build a list of account recovery options available for a successful password recovery.
-     *
-     * @param tenantDomain           Tenant domain
-     * @param recoveryInformationDTO RecoveryInformationDTO which wraps the password recovery information
-     * @return List of {@link AccountRecoveryType}
+    /*
+    * Return AccountRecoveryType object with recovery options available for the user.
+    * @param tenantDomain Tenant domain
+    * @param recoveryInformationDTO RecoveryInformationDTO which wraps the password recovery information
+    * @return AccountRecoveryType object with recovery options available for the user.
      */
-    private List<AccountRecoveryType> buildPasswordRecoveryInitResponse(String tenantDomain,
-                                                                        RecoveryInformationDTO recoveryInformationDTO) {
+    private AccountRecoveryType buildPasswordRecoveryInitResponse(String tenantDomain,
+                                                                  RecoveryInformationDTO recoveryInformationDTO) {
+
+        AccountRecoveryType accountRecoveryType = null;
+
+        if (recoveryInformationDTO == null) {
+            return accountRecoveryType;
+        }
```

Comment:
although this is functional, it can be improved. 

we could use a Optional return type here and do the early exit.

```
private Optional<AccountRecoveryType> buildPasswordRecoveryInitResponse(
    String tenantDomain,
    RecoveryInformationDTO recoveryInformationDTO) {

    if (recoveryInformationDTO == null) {
        return Optional.empty();
    }
```

or ... we could handle the recoveryInformationDTO == null scenario outside of the method. I prefer the Optional return type way.
WDYT?